client/my-sites/checkout/src/components/payment-method-step.tsx

```diff
@@ -59,7 +62,21 @@ export default function BeforeSubmitCheckoutHeader() {
 	const { responseCart } = useShoppingCart( cartKey );
 	const taxLineItems = getTaxBreakdownLineItemsFromCart( responseCart );
 	const creditsLineItem = getCreditsLineItemFromCart( responseCart );
+	const couponLineItem = getCouponLineItemFromCart( responseCart );
 	const shouldCollapseLastStep = useShouldCollapseLastStep();
+	const translate = useTranslate();
+	const subtotalWithoutCoupon =
+		responseCart.sub_total_integer + responseCart.coupon_savings_total_integer;
+	const subTotalLineItemWithoutCoupon: LineItemType = {
+		id: 'subtotal-without-coupon',
+		type: 'subtotal',
+		label: translate( 'Subtotal' ),
+		formattedAmount: formatCurrency( subtotalWithoutCoupon, responseCart.currency, {
+			isSmallestUnit: true,
+			stripZeros: true,
```

Comment:
Instead of building all the above here (and then separately building it in the other file), would it be possible to retain the call to `getSubtotalLineItemFromCart()` and put the coupon subtraction logic inside it?  Possibly renaming it to `getSubtotalLineItemWithoutCouponFromCart()` in the process.

As far as I can see, you've removed all the existing calls to `getSubtotalLineItemFromCart()` in this pull request already.

Reply:
True. I understand what you mean, but I think it's actually better to remove `getSubtotalLineItemFromCart` entirely. 

The reasoning is that the definition of "subtotal" is now in flux. The shopping cart considers a subtotal to be "the sum of each cart item's value without taxes", but this includes coupon discounts since they apply to each item and not to the cart as a whole. Since this PR changes the meaning of what's being displayed as the subtotal to "the sum of each cart item's value without taxes with coupon discounts removed", we now have two definitions of "subtotal" in calypso.

The "getXLineItemFromCart" functions are left over from when we had a `LineItem` type used by checkout to display each line item. That has all since been removed, but I left these functions because it was easier than refactoring them out. By not using them and by forcing the calculation to be explicit, it becomes easier to understand that what we're displaying isn't quite the same subtotal that the cart uses. 

That said, it would be nice to standardize this calculation so it can be used multiple places. I've merged in a new function to DRY the calculation. It's called `getSubtotalWithoutCoupon()` so it should be more clear what it does.